pkg/apiserver/registry/stats/nodelatencystat/rest.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodelatencystat
+
+import (
+	"context"
+	"fmt"
+
+	statsv1alpha1 "antrea.io/antrea/pkg/apis/stats/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metatable "k8s.io/apimachinery/pkg/api/meta/table"
+	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+// nodeLatencyCollector is the interface required by the handler.
+type nodeLatencyCollector interface {
+	Collect(summary *statsv1alpha1.NodeIPLatencyStat)
+	Get(name string) *statsv1alpha1.NodeIPLatencyStat
+	List() []statsv1alpha1.NodeIPLatencyStat
+}
+
+type REST struct {
+	nodeLatencyCollector nodeLatencyCollector
+}
+
+// nodeLatencyCollectorImpl implements nodeLatencyCollector.
+type nodeLatencyCollectorImpl struct {
+	nodeLatencyStats cache.Indexer
+	dataCh           chan *statsv1alpha1.NodeIPLatencyStat
```

Comment:
I don't think there is a need for a channel at all here? Why not update the `cache.Indexer` directly in `Create`?

I think you could also remove the `nodeLatencyCollector` interface and the `nodeLatencyCollectorImpl` struct, and overall make this file much simpler.

Reply:
That's a good suggestion, I'll remove that part.
The previous implementation was due to the need to initialize this cache indexer only once on the api server, so a struct was provided and initialized when the api server started.